inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java

```diff
@@ -635,11 +638,15 @@ private void parseFieldRelations(List<FieldInfo> fields,
             Map<String, FieldRelation> fieldRelationMap, StringBuilder sb) {
         for (FieldInfo field : fields) {
             FieldRelation fieldRelation = fieldRelationMap.get(field.getName());
+            FormatInfo fieldFormatInfo = field.getFormatInfo();
             if (fieldRelation == null) {
-                String targetType = TableFormatUtils.deriveLogicalType(field.getFormatInfo()).asSummaryString();
+                String targetType = TableFormatUtils.deriveLogicalType(fieldFormatInfo).asSummaryString();
                 sb.append("\n    CAST(NULL as ").append(targetType).append(") AS ").append(field.format()).append(",");
                 continue;
             }
+            boolean complexType = fieldFormatInfo instanceof RowFormatInfo
```

Comment:
Suggest extracting this check as a method into the `FieldInfoUtils`, then if we add a newly complex type, then we just modify `FieldInfoUtils` is enough, no need to modify this file.

Reply:
It's a good suggestion, but ```FieldInfoUtils.java``` is in ```manager-pojo``` and ```FlinkSqlParser.java``` is in ```sort-core```, I don't think it's suitable to import ```manager-pojo``` into ```sort-core```. Any other good idea for this? ^_^

Reply:
> Suggest extracting this check as a method into the `FieldInfoUtils`, then if we add a newly complex type, then we just modify `FieldInfoUtils` is enough, no need to modify this file.

It's a good suggestion, but ```FieldInfoUtils.java``` is in ```manager-pojo``` and ```FlinkSqlParser.java``` is in ```sort-core```, I don't think it's suitable to import ```manager-pojo``` into ```sort-core```. Any other good idea for this? ^_^

Reply:
Get it. I'll refactor it later.